### ISSUE TITLE
lerna clean --scope should respect --include-filtered-dependencies

### ISSUE THREAD

**Issue by roblg** [lerna/lerna#540]
This is phrased as a statement, but it's more of a question. In our internal repo, I just attempted to run:

```
➜  main git:(master) ✗ lerna clean --scope corvair --include-filtered-dependencies
Lerna v2.0.0-beta.34
Scoping to packages that match 'corvair'
About to remove the following directories:
- /home/roblg/code/main/corvair/node_modules
? Proceed? (ynH)
```

It picks up the one package, but not any of its dependencies. If I run `lerna bootstrap` with the same arguments, it finds all the relevant transitive dependencies (all 29):

```
➜  main git:(master) ✗ lerna bootstrap --scope corvair --include-filtered-dependencies
Lerna v2.0.0-beta.34
Scoping to packages that match 'corvair'
Bootstrapping 29 packages
...
```

Seems like `clean` should have the same behavior?

My setup:
```
lerna: v2.0.0-beta.34
node: 6.9.1
npm: 3.10.8
```

**Comment by gigabo** [comment]
Yeah, that would be handy.  Good suggestion @roblg.  Want to take it on? 😁

**Comment by roblg** [comment]
> Want to take it on?

No promises, but I might. :)

If anybody else wants it though, have at it...

**Comment by gpittarelli** [comment]
I'll take a stab at it tonight

**Comment by roblg** [comment]
@gpittarelli sorry, I should have posted here. I've actually already got it done locally. (Ended up wanting it for something at work sooner than I figured I would).

While I was looking at it though, it seems like there are a number of other commands that might benefit from respecting `--include-filtered-dependencies` (diff/exec/ls/run). Might even make sense to factor that out into `Command.js` or something so it's generally available (that's not my call though). I'm definitely _not_ planning on looking at `diff/exec/ls/run`, if you're interested in those.
